enterprise/server-enterprise/src/test/java/org/neo4j/server/rest/security/CypherRESTInteraction.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.server.rest.security;
+
+import org.codehaus.jackson.JsonNode;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.function.Consumer;
+import javax.ws.rs.core.HttpHeaders;
+
+import org.neo4j.graphdb.ResourceIterator;
+import org.neo4j.server.rest.domain.JsonHelper;
+import org.neo4j.server.rest.domain.JsonParseException;
+import org.neo4j.test.server.HTTP;
+
+import static org.junit.Assert.fail;
+import static org.neo4j.test.server.HTTP.RawPayload.quotedJson;
+
+class CypherRESTInteraction extends RESTInteraction
+{
+
+    @Override
+    String commitPath()
+    {
+        return "db/data/cypher";
+    }
+
+    CypherRESTInteraction( Map<String,String> config ) throws IOException
+    {
+        super( config );
+    }
+
+    @Override
+    public String executeQuery( RESTSubject subject, String call, Map<String,Object> params,
```

Comment:
Could rewrite this in the superclass and only extract the minor parts that are different.